### DISCUSSION TITLE
About bundled tcc: Is there an automatical script to build it?

### DISCUSSION THREAD

**Discussion by ZeroAurora** [vlang/v#15659] [General]
TBH I really love V's design and simplicity, so I've been following the development for quite a long time. But I don't actually trust the bundled tcc.
No I don't mean that V team will hack my computer, I just wonder if I could build all of these things from scratch, and importantly, in case that there is an attack on GitHub. (there is no signature on distributed binaries, after all!)
AFAIK the original tcc build for Windows doesn't work very well. And [this link](https://github.com/vlang/v/wiki/Installing-a-C-compiler-on-Windows#tcc) said that the bundled tcc is patched, but I don't know where the patch is.
Also, tcc is a project in development. So having a convenient way to build it can allow anyone to test new versions and to update distributed tcc more conveniently. And if a building script exists, the whole V toolchain, including compiler, becomes reproducible.
So would you please provide an automatical script for cloning, patching and building tcc?
Should be related to #12672 .

**Comment by JalonSolov** [comment]
The bundled tcc isn't actually patched, it is simply the latest commit at the time the binary was created.  So "patched" according to the official release, but nothing special other than using later code.

Check here for details on how to build:  https://github.com/vlang/tccbin

**Reply by ZeroAurora** [reply]
I tried to use the self-built tcc's headers with provided win32 headers manually, but failed because of the lack of specific headers. Is there a guide for that? Or should I open an issue?

**Reply by JalonSolov** [reply]
@spaceface777 ^^
